autofs/connectathon/cthon_automount/bin/setup

```diff
@@ -0,0 +1,241 @@
+#!/bin/bash
+#
+#ident	"$Id: setup,v 1.1 2005/10/25 22:38:34 jmoyer Exp $"
+#
+# Run this setup script to setup the automounter testing environment.
+#
+# setup [-a|-c] [-f server_name] [-s server_name] [-m mntpoint]
+# 	-a		perform complete setup. Need to be run as "root".
+#	-c		perform partial setup. Need to complete setup manually.
+#	-f server_name	server to failover to.
+#	-s server_name	server to test against.
+#	-m mntpoint	path to automount on locally
+# 
+# If no command line arguments are provided, then the environment variables 
+# must be defined in "../src/tests.init"
+#
+# This script installs all the necessary automounter maps in
+# ${AUTOMAP_DIR}. If automounter was not started automatically, you are 
+# responsible for manually modifying your /etc/auto_master map with the 
+# required changes (given by this script) and have your automounter mount 
+# the new mountpoints (by either restarting it, or running the "automount -v" 
+# command on autofs based systems.
+#
+# This script also creates the necessary directories on your server to
+# run the tests, provided ${AUTO_SERVER_DIR} is shared.
+#
+# This script also installs all the necessary data files for this testing
+# environment in ${AUTOMAP_DIR}/dat directory.
+#
+
+Progname=`basename $0`
+NSPC=`echo $Progname | sed 's/./ /g'`
+
+#
+# maps directory
+#
+mapDir=../maps
+
+MKDIR="mkdir"
+if [ -z "${MASTER_MAP}" ] ; then
+	MASTER_MAP=auto_master
+fi
+
+InitFile="../src/tests.init"
+USAGE="usage: $Program [-a|-c] [-f server_name] [-s server_name] [-m mntpoint]"
+SETUP=-a
+
+if [ ! -f ${InitFile} ]; then
+	echo "{Err} 'tests.init' not found - cannot proceed without defaults file"
+	exit 1
```

Comment:
This looks like a test setup issue, can we abort the task here instead ?

Reply:
@rasibley,
update cthon_setup()  in runttest.sh  to check return code of cthon_automount/bin/setup,
if not zero/successful,  the task will be aborted.

it will abort the task  when setup test return non zero (failed).